shuup/campaigns/models/campaigns.py

```diff
@@ -417,11 +426,6 @@ def has_been_used(self, usage_count=1):
         """ See if code is used the times given """
         return CouponUsage.objects.filter(coupon=self).count() >= usage_count
 
-    def save(self, **kwargs):
-        if Coupon.objects.filter(code__iexact=self.code, active=True).exclude(pk=self.pk).exists():
```

Comment:
Here the shops is not sharing the actual coupon. They are just sharing the code part of the coupon. The coupon is still onetoone-relation to the basket campaign.